Shared/Samples/Edit geodatabase with transactions/EditGeodatabaseWithTransactionsView.swift

```diff
@@ -0,0 +1,266 @@
+                    }
+                    
+                    Spacer()
+                    
+                    Toggle("Requires Transaction", isOn: $transactionIsRequired)
+                        .disabled(isInTransaction)
+                        .onChange(of: transactionIsRequired) { transactionIsRequired in
+                            statusText = transactionIsRequired
+                            ? "Tap Start to begin a transaction."
+                            : "Tap on the map to add a feature."
+                        }
+                    
+                    Spacer()
+                    
+                    Button("Sync") {
+                        selectedAction = .sync
+                    }
+                    .disabled(isInTransaction || !model.hasLocalEdits)
+                }
+            }
+            .alert("Commit Edits", isPresented: $endTransactionAlertIsPresented) {
+                Button("Commit") {
+                    selectedAction = .commitTransaction
+                }
+                Button("Rollback", role: .destructive) {
+                    selectedAction = .rollbackTransaction
+                }
+                Button("Cancel", role: .cancel) {}
+            } message: {
+                Text("Commit the edits in the transaction to the geodatabase or rollback to discard them.")
+            }
+            .disabled(selectedAction == .setUp || selectedAction == .sync)
+            .errorAlert(presentingError: $error)
+    }
+}
+
+/// An action associated with the geodatabase.
+private enum Action: Equatable {
+    /// Sets up the sample.
+    case setUp
+    /// Adds a feature of a given type to a table in the geodatabase.
+    case addFeature(tableName: String, featureTypeID: Int)
+    /// Starts a transaction on the geodatabase.
+    case beginTransaction
+    /// Commits the edits in the transaction to the geodatabase.
+    case commitTransaction
+    /// Rollback the edits in the transaction from the geodatabase.
+    case rollbackTransaction
+    /// Synchronizes the geodatabase and the feature service.
+    case sync
+    
+    /// The message to display when the action successfully completes.
+    var completionMessage: String {
+        switch self {
+        case .setUp: "Tap Start to begin a transaction."
+        case .addFeature: "Added feature."
+        case .beginTransaction: "Transaction started."
+        case .commitTransaction: "Edits committed to geodatabase."
+        case .rollbackTransaction: "Edits discarded."
+        case .sync: "Synchronized geodatabase and feature service."
+        }
+    }
+}
+
+/// A view allowing the user to select a feature type from given feature tables.
+private struct SelectFeatureTypeView: View {
+    /// The feature tables containing the feature types.
+    let featureTables: [ArcGISFeatureTable]
+    
+    /// The action to perform when the "Done" button is pressed.
+    let action: (_ tableName: String, _ featureTypeID: Int) -> Void
```

Comment:
Maybe use a more specific name, such as (I'm not particularly good at naming) `onFeatureSpecifiedAction`